actions/class.Campaign.php

```diff
@@ -1,70 +1,57 @@
 <?php
-/*  
+/**
  * This program is free software; you can redistribute it and/or
  * modify it under the terms of the GNU General Public License
  * as published by the Free Software Foundation; under version 2
  * of the License (non-upgradable).
- * 
+ *
  * This program is distributed in the hope that it will be useful,
  * but WITHOUT ANY WARRANTY; without even the implied warranty of
  * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
  * GNU General Public License for more details.
- * 
+ *
  * You should have received a copy of the GNU General Public License
  * along with this program; if not, write to the Free Software
  * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
- * 
+ *
  * Copyright (c) 2002-2008 (original work) Public Research Centre Henri Tudor & University of Luxembourg (under the project TAO & TAO2);
  *               2008-2010 (update and modification) Deutsche Institut für Internationale Pädagogische Forschung (under the project TAO-TRANSFER);
  *               2009-2012 (update and modification) Public Research Centre Henri Tudor (under the project TAO-SUSTAIN & TAO-DEV);
- * 
+ *               2012-2018 (original work) Open Assessment Technologies SA;
+ *
  */
-?>
-<?php
 
 use oat\tao\model\TaoOntology;
 
 /**
  * Campaign Controller provide actions performed from url resolution
- * 
+ *
  * @author CRP Henri Tudor - TAO Team - {@link http://www.tao.lu}
  * @package taoCampaign
- 
+
  * @license GPLv2  http://www.opensource.org/licenses/gpl-2.0.php
  */
- 
-class taoCampaign_actions_Campaign extends tao_actions_SaSModule {
-	
-	/**
-	 * constructor: initialize the service and the default data
-	 */
-	public function __construct(){
-		
-		parent::__construct();
-		
-		//the service is initialized by default
-		$this->service = taoCampaign_models_classes_CampaignService::singleton();
-		$this->defaultData();
-		
-	}
-	
-	protected function getClassService(){
-		return taoCampaign_models_classes_CampaignService::singleton();
-	}
-	
-/*
- * controller actions
- */
+
+class taoCampaign_actions_Campaign extends tao_actions_SaSModule
+{
+    protected function getClassService()
+    {
+        if (is_null($this->service)) {
+            $this->service = taoCampaign_models_classes_CampaignService::singleton();
+        }
+        return $this->service;
+    }
+
 	/**
-	 * Render json data to populate the campaign tree 
+	 * Render json data to populate the campaign tree
 	 * 'modelType' must be in the request parameters
 	 */
-	public function getCampaigns(){
-		
-		if(!tao_helpers_Request::isAjax()){
+	public function getCampaigns()
+    {
```

Comment:
Missing calling:
`$this->defaultData();` in `getCampaigns` , `addCampaign`, `addCampaignClass`, `delete`, `cloneCampaign`, `getDeliveries`